realm/realm-library/src/androidTest/kotlin/io/realm/DictionaryTests.kt

```diff
@@ -296,6 +306,108 @@ class DictionaryTests {
         // TODO
     }
 
+    @Test
+    fun managed_put() {
+        realm.executeTransaction { transactionRealm ->
+            transactionRealm.copyToRealm(initDictionaryClass())
+
+            val instance = realm.where<DictionaryClass>()
+                    .findFirst()
+            assertNotNull(instance)
+
+            with(instance) {
+                // All types but byte[] can be asserted like this
+                myBooleanDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO)
+                myStringDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_STRING)
+                myIntegerDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC)
+                myFloatDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC.toFloat())
+                myLongDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC.toLong())
+                myShortDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC.toShort())
+                myDoubleDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC.toDouble())
+                myByteDictionary!!.putAndAssert(KEY_HELLO, VALUE_HELLO_NUMERIC.toByte())
+                myDateDictionary!!.putAndAssert(KEY_HELLO, Date())
+                myObjectIdDictionary!!.putAndAssert(KEY_HELLO, ObjectId())
+                myUUIDDictionary!!.putAndAssert(KEY_HELLO, UUID.randomUUID())
+                myDecimal128Dictionary!!.putAndAssert(KEY_HELLO, Decimal128(42))
+
+                val previousValue = myByteArrayDictionary!!.put(KEY_HELLO, VALUE_HELLO_BYTE_ARRAY)
+                assertNull(previousValue)
+                val actual = myByteArrayDictionary!![KEY_HELLO]
+                assertEquals(Byte.MIN_VALUE, actual!![0])
+                assertEquals(Byte.MAX_VALUE, actual[1])
+            }
+        }
+    }
+
+    @Test
+    fun managed_get() {
+        realm.executeTransaction { transactionRealm ->
+            transactionRealm.copyToRealm(initDictionaryClass(true))
+        }
+
+        val instance = realm.where<DictionaryClass>()
+                .findFirst()
+        assertNotNull(instance)
+
+        with(instance) {
```

Comment:
Would it somehow be possible to refactor these tests to use enum switches like our other Collection tests? There are so many supported datatypes that it is very easy to forget some, so using an enum makes sure we catch all of them, e.g. like https://github.com/realm/realm-java/blob/master/realm/realm-library/src/androidTest/java/io/realm/RealmResultsTests.java#L813

This is especially important later if we add a new datatype, as figuring out all the the tests that need to change is pretty hard.

Reply:
Yes, I created a ticket for that. I will address that in one of the upcoming PRs.